core/crdt/composite.go

```diff
@@ -154,13 +154,26 @@ func (c CompositeDAG) Merge(ctx context.Context, delta core.Delta, id string) er
 		return c.deleteWithPrefix(ctx, c.key.WithValueFlag().WithFieldId(""))
 	}
 
-	// ensure object marker exists
-	exists, err := c.store.Has(ctx, c.key.ToPrimaryDataStoreKey().ToDS())
+	// We cannot rely on the dagDelta.Status here as it may have been deleted locally, then an update
+	// commit synced via P2P.  The synced update commit should not undelete the local.
+	versionKey := c.key.WithValueFlag().WithFieldId(core.DATASTORE_DOC_VERSION_FIELD_ID)
+	objectMarker, err := c.store.Get(ctx, c.key.ToPrimaryDataStoreKey().ToDS())
+	hasObjectMarker := !(err != nil && errors.Is(err, ds.ErrNotFound))
+	if err != nil && hasObjectMarker {
```

Comment:
This looks very strange - thanks for flagging.  I'll change it for sure.

- [ ] change weird composite.go code

Reply:
hasObjectMarker is actually read twice, and the value the the name provides is pretty significant on the second read.  Leaving as-is.

Reply:
You could at least change it to `hasObjectMarker := !errors.Is(err, ds.ErrNotFound)`

Reply:
The `error.Is` documentation (on the declaration) does not state what the behaviour is if you pass in a nil `err`

Reply:
I'm not sure it needs to state it specifically. `nil` is certainly not the same as `ds.ErrNotFound`.

Reply:
`nil`s like to panic, and I dont like passing them about if there is not a really really good reason to.

Reply:
I don't think a panic is a problem here. `nil` errors are a normal thing in go.

Reply:
For example: https://go.dev/play/p/XSjvPih2ek3

Reply:
My last comment was not directed at just nil errors, but any nil param

Reply:
Do feel that it is sensible to only check for nil depending on the type of object that the pointer *would* have pointed to were it actually pointing to anything, as opposed to either always checking, or always assuming it would be fine and existing?

Reply:
This comes up often, `nil` is completely fine in Go to pass around, especially for errors. ALL error code accounts for `nil` error values for the average Go dev. Moreover, the standard library is VERY safe to do this on. Please do the single `Is` call :)

Reply:
Additionally, its very clear (imo) what `!errors.Is(err, ds.ErrNotFound0` is trying to say, without it being assigned to a named variable. Less concerned about this point, but please do take care of the other comment 👍

Reply:
If there is a desire to close out this conversation:

You guys see this as a PR-critical blocker, and not idle chit-chat about a meaningless developer preference?

Reply:
I would say it's an unnecessarily complex line that makes the reader pause to double check what is really happening and the extra complexity adds no benefit. So no it's not idle chit-chat about meaningless developer preference.

Reply:
Fair enough, will change

- [ ] Remove nil check